aws-janitor/resources/internet_gateways.go

```diff
@@ -77,36 +74,46 @@ func (InternetGateways) MarkAndSweep(opts Options, set *Set) error {
 				break
 			}
 
-			var publicIPsToRelease []*string
+			if opts.DisassociatePublicIP {
```

Comment:
I am really not sure we need to keep this code - addresses are supposed to be disassociated when higher level resources are deleted, e.g. VMs or ELBs. If there would be addresses that warrant explicit deletion, it would likely be part of the code base, and earlier observed  IGW deletion issues are likely due to untagged ELBs that end up outliving IGWs and holding addresses associated.

We probably could roll back commit that introduced this and get rid of additional complexity.

Reply:
I see, I can close this and send the revert one: https://github.com/kubernetes-sigs/boskos/pull/163